### ISSUE TITLE
Calling done() multiple time inside a single test scope leads to unpredictable behavior

### ISSUE THREAD

**Issue by janvanderhaegen** [jasmine/jasmine#822]
I've started using Jasmine 2.1.3 for some async tests, and found the done() a really helpful future.
In one of my tests, I accidentely called done() twice:

```javascript
describe('Some random async tests', () => {
  it('should not be passing', (done) => {
    //done();
    setTimeout( () => {
      expect(true).toBe(false);
      done();
    },
    200
    );
  });

  it('should be passing1',(done) => {
    setTimeout(
      () => { done(); },
      200
    )
  });
  it('should be passing2',(done) => {
    setTimeout(
      () => { done(); },
      200
    )
  });
  it('should be passing3',(done) => {
    setTimeout(
      () => { done(); },
      200
    )
  }); 
);
```
When in 'it should not be passing', the first done() is commented out, things work as expected: 3/4 tests succeed and the failing test is 'it should not be passing'.
However, if you allow the 'it should not be passing' to call done(), then later (timeout) make an invalid expect and call done again, the test runner reports 3/4 tests failed and the failing test is 'it should be passing2'.

I've debugged my tests for countless hours and doubted my entire code base as every time I commented out any 'it should be passing' test that was failing, one other test that was working before would now randomly start failing.

Agreed, developers should not call done() multiple times in a single test, but when they do it would be nicer to inform them of their mistake than to report a false negative on another random test.

**Comment by slackersoft** [comment]
This looks like it is more likely the result of doing more work in your spec after calling `done` than that you've called it twice. The failure mode that we saw in 2.0.0, before it was fixed in 2.0.1, was that calling `done` a second time, would start running the next spec again.

Here I think you would see the same behavior you're seeing now even if you remove the call to `done` in the first `setTimeout` as long as you call `done` before `setTimeout`.

Unfortunately, there isn't much we can do about this, especially in the async case. Once you've called `done` you've told jasmine that it is safe to move on to the next thing in the queue.

Closing. Thanks for using jasmine!
